src/fileformat/types/import_table/elf_import_table.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file include/retdec/fileformat/types/import_table/elf_import_table.h
+ * @brief Class for ELF import table.
+ * @copyright (c) 2021 Avast Software, licensed under the MIT license
+ */
+
+#include "retdec/fileformat/utils/crypto.h"
+#include "retdec/utils/string.h"
+#include "retdec/fileformat/utils/tlsh/tlsh.h"
+#include "retdec/fileformat/types/import_table/elf_import_table.h"
+#include <algorithm>
+#include <regex>
+
+using namespace retdec::utils;
+
+namespace retdec {
+namespace fileformat {
+
+static const std::set<std::string> exclusion_set = {
+	"__libc_start_main", // main function
+	"main", // main function
+	"abort", // ARM default
+	"cachectl", // MIPS default
+	"cacheflush", // MIPS default
+	"puts", // Compiler optimization (function replacement)
+	"atol", // Compiler optimization (function replacement)
+	"malloc_trim" // GNU extensions
+};
+
+bool isSymbolExcluded(std::string symbol)
+{
+	if (symbol.empty()) {
+		return true;
+	}
+	/* ignore:
+		symbols starting with . or _
+		x86-64 specific functions
+		string functions (str.* and mem.*), gcc changes them depending on architecture
+		symbols starting with . or _
+	*/
+	std::regex exclusion_regex("(^[_\.].*$)|(^.*64$)|(^str.*$)|(^mem.*$)");
```

Comment:
Again, how/why/where was this get? What if somebody wanted to modify it?

Reply:
Also, with regexes one always have to think when and how many times they get compiled. I always have to google it myself, so I will leave this one for you - read on it and think about it. Maybe I'm wrong, but I think this regexe could be made `static`.